O/onnxruntime/build_tarballs.jl

```diff
@@ -0,0 +1,83 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "onnxruntime"
```

Comment:
This recipe will add `onnxruntime` JLL - and provide the CPU-only `libonnxruntime` LibraryProduct.

As mentioned in https://github.com/jw3126/ONNXRunTime.jl/issues/12, subsequent recipes will add execution providers, cf. https://onnxruntime.ai/docs/build/eps.html

Only some of these EP's are built as shared libraries on top of the CPU-only `libonnxruntime`: https://onnxruntime.ai/docs/build/eps.html#execution-provider-shared-libraries

Is the following sensible?

* `onnxruntime_cuda` to be defined in `O/onnxruntime/onnxruntime_cuda/build_tarballs.jl` will depend on `CUDNN` and provide library products `libonnxruntime` (CUDNN-dependent) and `libonnxruntime_providers_cuda`.
* `onnxruntime_providers_tensorrt` to be defined in `O/onnxruntime/onnxruntime_providers_tensorrt/build_tarballs.jl` will depend on `onnxruntime_cuda` and add `libonnxruntime_providers_tensorrt`.
* `onnxruntime_providers_onednn` to be defined in `O/onnxruntime/onnxruntime_providers_onednn/build_tarballs.jl` will depend on `onnxruntime` and add `libonnxruntime_providers_onednn`.

and so on for e.g. CoreML, AMD MiGraphX.

Or should `onnxruntime_cuda` be `onnxruntime_providers_cuda`?

Might it be an idea to include the CUDA-part in this PR (as it seems the cuda `libonnxruntime` will be independent of the CPU `libonnxruntime`)?

Reply:
Look superficially ok, although I'm not very into the GPU side of the libraries.  I know they're pretty complicated to deal with and in most cases I believe we can only build for x86_64-linux-gnu (because `nvcc` isn't a cross-compiler).  If you plan to add other sibling projects you may want to move this now to `O/ONNXRuntime/ONNXRuntime/build_tarballs.jl` (or whatever name you choose to use in the end)

Reply:
Yes, it seems we have to stick with re-packaging binaries for the CUDA-parts for now.

I have renamed to `O/ONNXRuntime/ONNXRuntime/build_tarballs.jl`.